File: frontend/src/types/auto-subscribe.ts
/**
 * 自动订阅相关类型 — 镜像 backend/app/schemas/auto_subscribe.py。
 */

/** 自动订阅配置（前端维护的完整字段集） */
export interface AutoSubConfig {
  enabled: boolean
  min_year: number
  min_vote: number
  media_type: string
  schedule_cron: string
  douban_enabled: boolean
  douban_ranks: string[]
  douban_rsshub: string
  douban_rss_custom: string
  mikan_enabled: boolean
  mikan_season: string
  mikan_year: number
  mikan_resolve_detail: boolean
  maoyan_enabled: boolean
  maoyan_movie_box: boolean
  maoyan_web_platforms: string[]
  maoyan_web_types: string[]
  maoyan_num: number
  proxy_url: string
  [key: string]: unknown
}

/** 自动订阅配置响应 */
export interface AutoSubConfigResponse {
  config: Partial<AutoSubConfig>
  last_run: string | null
  last_stats: Record<string, unknown> | null
  enabled_sources: string[]
  status_labels: Record<string, string>
  source_names: Record<string, string>
  running: boolean
  last_error: string | null
}

/** 自动订阅配置更新请求 */
export interface AutoSubConfigUpdate {
  config: Partial<AutoSubConfig>
}

/** 手动触发运行响应 */
export interface AutoSubRunResponse {
  success: boolean
  started: boolean
  message: string
}

/** 单条自动订阅历史记录 */
export interface AutoSubHistoryItem {
  key: string
  status: string
  time: string | null
  tmdb_id: number | null
  media_type: string | null
}

/** 自动订阅历史响应 */
export interface AutoSubHistoryResponse {
  items: AutoSubHistoryItem[]
  last_run: string | null
  stats: Record<string, unknown> | null
}

/** 下拉选项 */
export interface SelectOption {
  value: string
  label: string
}

/** 自动订阅元数据响应（供前端下拉选项） */
export interface AutoSubMetaResponse {
  douban_ranks: SelectOption[]
  maoyan_platforms: SelectOption[]
  maoyan_media_types: SelectOption[]
  seasons: SelectOption[]
  source_names: Record<string, string>
  status_labels: Record<string, string>
}

File: backend/app/schemas/auto_subscribe.py
"""自动订阅 Pydantic schemas。"""

from pydantic import BaseModel, Field


class AutoSubConfigResponse(BaseModel):
    """自动订阅配置响应。"""

    config: dict = Field(default_factory=dict)
    last_run: str | None = None
    last_stats: dict | None = None
    enabled_sources: list[str] = Field(default_factory=list)
    status_labels: dict = Field(default_factory=dict)
    source_names: dict = Field(default_factory=dict)
    running: bool = False
    last_error: str | None = None


class AutoSubConfigUpdate(BaseModel):
    """自动订阅配置更新请求。"""

    config: dict = Field(default_factory=dict)


class AutoSubRunResponse(BaseModel):
    """手动触发运行响应。"""

    success: bool = True
    started: bool = True
    message: str = ""


class AutoSubHistoryItem(BaseModel):
    """单条自动订阅历史记录。"""

    key: str
    status: str
    time: str | None = None
    tmdb_id: int | None = None
    media_type: str | None = None


class AutoSubHistoryResponse(BaseModel):
    """自动订阅历史响应。"""

    items: list[AutoSubHistoryItem] = Field(default_factory=list)
    last_run: str | None = None
    stats: dict | None = None


class AutoSubMetaResponse(BaseModel):
    """自动订阅元数据响应（供前端下拉选项）。"""

    douban_ranks: list[dict] = Field(default_factory=list)
    maoyan_platforms: list[dict] = Field(default_factory=list)
    maoyan_media_types: list[dict] = Field(default_factory=list)
    seasons: list[dict] = Field(default_factory=list)
    source_names: dict = Field(default_factory=dict)
    status_labels: dict = Field(default_factory=dict)
